### ISSUE TITLE
Pale Moon Browser App.JS

### ISSUE THREAD

**Issue by armouredking** [BookStackApp/BookStack#3368]
### Describe the Bug

Website javascript features from app.js do not function (such as sorting books).

### Steps to Reproduce

1. Use Pale Moon recent browser versions
2. Login to website with an account (if private)
3. Attempt to sort books with the upper right hand menu. NOTE: this is not about sorting pages, chapters, or book's orders. That works to an extent (Show Other Books when sorting book contents just displays loading animation).

### Expected Behaviour

Menus function as they do in other browsers.

### Screenshots or Additional Context

SyntaxError: expected expression, got '.'

Tested working with MSEdge, Brave, and Chrome.

### Browser Details

PaleMoon 29.4.5 and 29.4.5.1 on Windows 11

### Exact BookStack Version

v22.03.1

### PHP Version

LinuxServer Docker

### Hosting Environment

LinuxServer Docker Container

**Comment by ssddanbrown** [comment]
Thanks @armouredking, Can confirm this from a quick test on my side too.

It seems to be due to some modern JavaScript syntax usages (I've seen optional chaining and dynamic imports from my testing) which are not supported in pale moon (Both known issues in the pale moon projects, though there may be other incompatibilities). 

I'd be happy to go out my way if these were very new features, as I generally ensure a buffer to support other general browser-lag scenarios such as FireFox ESR usage, but the features I've come across are things that landed in FireFox 2+ years ago. I'd also be happy to go out my way if this was breaking due to anything vendor specific, but these elements currently causing issues are general web standards.

If there are very minor tweaks that can be made to achieve compatibility, I'd be happy to receive and review a PR, otherwise Pale Moon is not something I'd generally be testing on or assure support for.

**Comment by armouredking** [comment]
Thing is, I know it worked in Pale Moon previously. I'm not sure if it was a javascript update, deprecation, or the like, or perhaps the PM direction.

Originally, I was running Bookstack on an LXC and ran across this as I moved servers (moved over files and ENV from the LXC to the Docker instances and then updated the databases). I had not stayed on top of my updates when it was running from LXC, so I know this issue does not exist with v21 of Bookstack via the LXC.

**Comment by ssddanbrown** [comment]
@armouredking Sure, I don't doubt that it did work in Pale Moon, it would be very likely that we'd have started to use those features (Or upped or JavaScript compatibility output levels) within the last 2-12 months. It's simply that we've introduced some more modern features into our JavaScript output that Pale Moon does not yet support.

The features lacking support I've noticed are tracked on Pale Moon's bug tracker though:
- https://repo.palemoon.org/MoonchildProductions/UXP/issues/1658
- https://repo.palemoon.org/MoonchildProductions/UXP/issues/1691
But they look like tricky features to support.
